src/wallet/wallet.h

```diff
@@ -1180,6 +1180,11 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
 
     /** Whether a given output is spendable by this wallet */
     bool OutputEligibleForSpending(const COutput& output, const CoinEligibilityFilter& eligibility_filter) const;
+
+    /** mutex, container and condition variable for updating the latest transaction id */
+    std::mutex m_cs_txadd;
+    std::condition_variable m_cond_txadd;
+    uint256 m_latest_wtxid;
```

Comment:
Could keep `wtxOrdered` back iterator instead:
```
TxItems::const_iterator m_latest_wtx = wtxOrdered.rbegin();
```
Then above could be:
```cpp
if (m_latest_wtxid != wtxOrdered.rbegin()) {
    ...
}

Reply:
I prefer to keep a non references object in this case.